### PR TITLE
Add option to disable synonyms management in Magento

### PR DESCRIPTION
**Pull request by JanPetr**
Implements #577

### CODE REVIEW

File: app/code/community/Algolia/Algoliasearch/etc/system.xml [hunk 1]
```diff
@@ -648,6 +648,20 @@
...
+                            <show_in_website>1</show_in_website>
+                            <show_in_store>1</show_in_store>
+                            <comment>
+                                <![CDATA[
+                                If set to No, synonyms will be managed only in Algolia's dashboard and won't get overriden by Magento
+                                ]]>
+                            </comment>
+                        </enable_synonyms>
                         <synonyms translate="label comment">
                             <label>Synonyms</label>
```
Comment (maxiloc) [inline app/code/community/Algolia/Algoliasearch/etc/system.xml]: I would add a dependency for other synonyms settings, to not display when synonyms is False

File: js/algoliasearch/internals/adminhtml/admin_scripts.js [hunk 2]
```diff
@@ -54,4 +54,24 @@ algoliaAdminBundle.$(function($) {
 			return parseInt(n, 10);
 		});
 	}
+
```
Comment (maxiloc) [inline js/algoliasearch/internals/adminhtml/admin_scripts.js]: All that should be handled by the dependency

File: app/code/community/Algolia/Algoliasearch/etc/system.xml (hunk 1 above)
Comment (JanPetr) [inline app/code/community/Algolia/Algoliasearch/etc/system.xml]: I did that. But it does not work on `synonyms` and `one-way synonyms`, it works only on `synonyms file`. I don't know why, probably Magento's bug. I wrote own script for that: https://github.com/algolia/algoliasearch-magento/pull/585/files#diff-a7e898950149f6c176346608d6d5cba0R58

File: js/algoliasearch/internals/adminhtml/admin_scripts.js (hunk 2 above)
Comment (JanPetr) [inline js/algoliasearch/internals/adminhtml/admin_scripts.js]: See https://github.com/algolia/algoliasearch-magento/pull/585#discussion_r78739478 :)

**Pull request merged**